extensions/adapters/vector/src/main/java/mil/nga/giat/geowave/adapter/vector/utils/GeometryUtils.java

```diff
@@ -0,0 +1,236 @@
+package mil.nga.giat.geowave.adapter.vector.utils;
+
+import java.awt.geom.Point2D;
+import java.util.Arrays;
+import java.util.Collections;
+
+import javax.measure.unit.SI;
+import javax.measure.unit.Unit;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.log4j.Logger;
+import org.geotools.ows.bindings.UnitBinding;
+import org.geotools.referencing.GeodeticCalculator;
+import org.opengis.referencing.crs.CoordinateReferenceSystem;
+import org.opengis.referencing.cs.CoordinateSystem;
+import org.opengis.referencing.operation.TransformException;
+
+import com.vividsolutions.jts.geom.Coordinate;
+import com.vividsolutions.jts.geom.Envelope;
+import com.vividsolutions.jts.geom.Geometry;
+import com.vividsolutions.jts.geom.GeometryFactory;
+import com.vividsolutions.jts.geom.LineString;
+import com.vividsolutions.jts.geom.Point;
+
+public class GeometryUtils
+{
+
+	private static Logger LOGGER = Logger.getLogger(GeometryUtils.class);
+
+	/**
+	 * Build a buffer around a geometry for the pro
+	 * 
+	 * @param crs
+	 * @param geometry
+	 * @param distanceUnits
+	 * @param distance
+	 * @return
+	 * @throws TransformException
+	 */
+	public static final Pair<Geometry, Double> buffer(
+			final CoordinateReferenceSystem crs,
+			final Geometry geometry,
+			final String distanceUnits,
+			final double distance )
+			throws TransformException {
+		Unit<?> unit;
+		try {
+			unit = (Unit<?>) new UnitBinding().parse(
+					null,
+					distanceUnits);
+		}
+		catch (final Exception e) {
+			unit = SI.METER;
+			LOGGER.warn(
+					"Cannot lookup unit of measure " + distanceUnits,
+					e);
+		}
+		final double meterDistance = unit.getConverterTo(
+				SI.METER).convert(
+				distance);
+		final double degrees = distanceToDegrees(
+				crs,
+				geometry,
+				meterDistance);
+		// buffer does not respect the CRS; it uses simple cartesian math.
+		return Pair.of(
+				adjustGeo(
```

Comment:
If I do not keep this adjustment code, then the geometries produced by buffer on invalid. This code is focused fixing the coordinate space issue. Thus, I think this needs to stay.  

That being said, the other issue is the wrapping problem you are referring to.  The problem will likely arise when performing the intersection (after the the DWITHIN conversion).
I agree that cutting into a multi-polygon is  the best option. My thought is to have another function take the wrapped geometry and convert, using that method when converting DWITHIN to intersection.